src/Text/Parsing/Parser/Pos.purs

```diff
@@ -0,0 +1,35 @@
+module Text.Parsing.Parser.Pos where
+
+import Data.String (split)
+import Data.Foldable (foldl)
+
+-- | `Position` represents the position of the parser in the input.
+-- | - `line` is the current line in the input
+-- | - `column` is the column of the next character in the current line that will be parsed
+data Position = Position
+  { line :: Number
+  , column :: Number
+  }
+
+instance showPosition :: Show Position where
+  show (Position { line: line, column: column }) =
+    "(line " ++ show line ++ ", column " ++ show column ++ ")"
```

Comment:
Can this change to generate executable code, and have a separate function for error rendering?